src/Datadog.Trace/Tagging/MsmqTags.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Datadog.Trace.ExtensionMethods;
+
+namespace Datadog.Trace.Tagging
+{
+    internal class MsmqTags : InstrumentationTags
+    {
+        protected static readonly IProperty<string>[] MsmqTagsProperties =
+           InstrumentationTagsProperties.Concat(
+               new Property<MsmqTags, string>(Trace.Tags.InstrumentationName, t => t.InstrumentationName, (t, v) => t.InstrumentationName = v),
+               new Property<MsmqTags, string>(Trace.Tags.MsmqCommand, t => t.Command, (t, v) => t.Command = v),
+               new Property<MsmqTags, string>(Trace.Tags.MsmqQueue, t => t.Queue, (t, v) => t.Queue = v),
+               new Property<MsmqTags, string>(Trace.Tags.MsmqQueueLabel, t => t.QueueLabel, (t, v) => t.QueueLabel = v),
+               new Property<MsmqTags, string>(Trace.Tags.MsmqQueueLastModifiedTime, t => t.QueueLastModifiedTime, (t, v) => t.QueueLastModifiedTime = v),
+               new Property<MsmqTags, string>(Trace.Tags.MsmqQueueUniqueName, t => t.UniqueQueueName, (t, v) => t.UniqueQueueName = v),
+               new Property<MsmqTags, string>(Trace.Tags.MsmqIsTransactionalQueue, t => t.IsTransactionalQueue, (t, v) => t.IsTransactionalQueue = v),
+               new Property<MsmqTags, string>(Trace.Tags.MsmqTransactionType, t => t.TransactionType, (t, v) => t.TransactionType = v));
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MsmqTags"/> class.
+        /// </summary>
+        /// <param name="spanKind">kind of span</param>
+        public MsmqTags(string spanKind) => SpanKind = spanKind;
+
+        public string Command { get; set; }
+
+        /// <inheritdoc/>
+        public override string SpanKind { get; }
+
+        public string Queue { get; set; }
+
+        public string QueueLabel { get; set; }
+
+        public string QueueLastModifiedTime { get; set; }
+
+        public string UniqueQueueName { get; set; }
+
+        public string TransactionType { get; set; }
+
+        public string IsTransactionalQueue { get; set; }
+
+        public string InstrumentationName { get; set; }
```

Comment:
This is a lot of tags, do we definitely want them all? I've learnt (the hard way!) that each tag has a significant performance impact! IMO, we could probably lose:

- `QueueLabel`
-  `QueueLastModifiedTime`
- `IsTransactionalQueue`
- `UniqueQueueName` (i.e. either keep `formatName` or `queueName`, depending on which is more useful!)
- `TransactionType`

Additionally `InstrumentationName` is normally created as a `readonly` field in the Tags, rather than specified at creation 🙂

Reply:
Oh ok good to know! will remove those and keep format which gives the host info.

In msmq, if we send a message within a transaction to a non transactional queue, or the opposite, sending a message outside of a transaction context to a transactional queue, the message is never sent but the methods are called and nothing crashes, which is kind of an odd behavior I got stuck on. So I thought it might be useful to a customer to see, in the operation msmq.send, for example: 

is_transactional_queue > False
transaction_type> Single (there is Auto, None, Single)

but it's just a remark, I'm not sure of their requirements here :)

Reply:
If unique name includes Path (as it seems to from that example), then that will probably be sufficient IMO. (Spoken as someone that has never used MSMQ!)